govcd/catalog.go

```diff
@@ -360,6 +360,49 @@ func (cat *Catalog) UploadOvfByLink(ovfUrl, itemName, description string) (Task,
 	return task, nil
 }
 
+// CaptureVappTemplate captures a vApp template from an existing vApp
+func (cat *Catalog) CaptureVappTemplate(captureParams *types.CaptureVAppParams) (*VAppTemplate, error) {
+	task, err := cat.CaptureVappTemplateAsync(captureParams)
+	if err != nil {
+		return nil, err
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, err
+	}
+
+	// After the task is finished, owner field contains the resulting vApp template
+	return cat.GetVappTemplateByHref(task.Task.Owner.HREF)
```

Comment:
Should we test for `task.Task.Owner` to exist before returning?
We risk a panic if the task had an accident and lost the owner

Reply:
Patched and ran test